### PR TITLE
Redirect to different post-register set up page, with the new user logged in

### PR DESCRIPTION
**Pull request by francisli**

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -40,9 +39,14 @@ function App() {
...
           </Route>
           {process.env.REACT_APP_FEATURE_REGISTRATION === 'true' && (
-            <Route path="/register">
-              <Register />
-            </Route>
+            <>
+              <Route path="/register">
+                <Register />
+              </Route>
+              <AuthProtectedRoute path="/setup">
```
Comment (francisli) [inline client/src/App.js:46]: AuthProtectedRoute is used for routes that should only be accessible when logged in.

File: client/src/Register.js [hunk 2]
```diff
@@ -8,16 +8,18 @@ import UnexpectedError from './UnexpectedError';
 import ValidationError from './ValidationError';
 
 import './Register.scss';
+import { useAuthContext } from './AuthContext';
 
 function Register() {
+  const authContext = useAuthContext();
   const history = useHistory();
 
+  const [role, setRole] = useState('donor')
```
Comment (francisli) [inline client/src/Register.js:17]: Currently, there is no role attribute in user. So, I created a separate state variable for tracking it, and set an initial value of "donor"...

File: client/src/Register.js [hunk 3]
```diff
@@ -31,8 +33,9 @@ function Register() {
     event.preventDefault();
     setError(null);
     try {
-      await Api.auth.register(user);
-      history.push('/login', { flash: 'Your account has been created!' });
+      const response = await Api.auth.register(user);
+      authContext.setUser(response.data);
+      history.push(`/setup/${role}`, { flash: 'Your account has been created!' });
```
Comment (francisli) [inline client/src/Register.js:38]: Here I change the next page after registration to be based on the role.

File: client/src/Register.js [hunk 4]
```diff
@@ -31,8 +33,9 @@ function Register() {
     event.preventDefault();
     setError(null);
     try {
-      await Api.auth.register(user);
-      history.push('/login', { flash: 'Your account has been created!' });
+      const response = await Api.auth.register(user);
+      authContext.setUser(response.data);
```
Comment (francisli) [inline client/src/Register.js:37]: Here we take the new user returned by the server after registration and set it in the Auth context, which makes it available throughout the site as the logged in user.

File: client/src/Register.js [hunk 5]
```diff
@@ -59,12 +62,10 @@ function Register() {
               )}
               <div className="form-group">
                 <label for="RoleSelect"><strong>Role</strong></label>
-                <select className="form-control">
-                  <option selected="selected">Select Role</option>
-                  <option> Non-Profit Partner</option>
-                  <option> Program Director </option>
-                  <option> Student </option>
-                  <option> Donor </option>
+                <select className="form-control" value={role} onChange={(event) => setRole(event.target.value)}>
```
Comment (francisli) [inline client/src/Register.js:65]: Here we add the props necessary for the dropdown to read/set the role

File: client/src/Setup/Setup.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import Donor from './Donor';
+import NonProfit from './NonProfit';
+import ProgramDirector from './ProgramDirector';
+
+function Setup() {
+  const {path, url} = useRouteMatch();
+
+  return (
+    <Switch>
+      <Route path={`${path}/donor`}>
```
Comment (francisli) [inline client/src/Setup/Setup.js:12]: Here I've set up placeholders for the next step in registration.

File: routes/api/auth.js [hunk 7]
```diff
@@ -57,7 +57,10 @@ if (process.env.REACT_APP_FEATURE_REGISTRATION === 'true') {
       await user.save();
       await user.hashPassword(user.password);
       await user.sendWelcomeEmail();
-      res.status(HttpStatus.CREATED).json(user);
+      req.login(user, function(err) {
```
Comment (francisli) [inline routes/api/auth.js:60]: This ensures the user is logged in (and a cookie set) after registration.

**Review by cesar-ca — APPROVED** [review]
Works on my end


https://user-images.githubusercontent.com/44204337/118811192-91a85a00-b861-11eb-8d7a-2697ec220dc3.mov

**Pull request merged**